irohad/simulator/impl/simulator.cpp

```diff
@@ -112,6 +118,11 @@ namespace iroha {
       log_->info("process verified proposal");
 
       const auto &proposal = verified_proposal_and_errors->verified_proposal;
+
+      log_->trace("Process verified proposal: [ {} ]",
+                  shared_model::interface::TxHashesPrinter<decltype(
+                      proposal->transactions())>(proposal->transactions()));
```

Comment:
You can do the same in all the similar places
```
      const auto &txs = proposal->transactions();
      log_->trace("Process verified proposal: [ {} ]",
                  shared_model::interface::TxHashesPrinter<decltype(txs)>(txs));
```

Reply:
@x3medima17 sorry for writing a patch for your PR again, but I think there is a better way to avoid explicit template arguments:
[0001-makeTxHashesPrinter-helper-func.patch.txt](https://github.com/hyperledger/iroha/files/3442699/0001-makeTxHashesPrinter-helper-func.patch.txt)